### PR TITLE
Add 0.14.0 Release Notes

### PR DESCRIPTION
**Pull request by irq0**
# Describe your changes

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] CHANGELOG.md has been updated should there be relevant changes in this PR.

### CODE REVIEW

File: docs/release-notes/s3gw-v0.14.0.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Release Notes - v0.13.0
```
Comment (m-ildefons) [inline docs/release-notes/s3gw-v0.14.0.md]: Isn't this supposed to be `v0.14.0`?

**Review by m-ildefons — CHANGES_REQUESTED** [review]
Wrong version number in title.

File: docs/release-notes/s3gw-v0.14.0.md (hunk 1 above)
Comment (irq0) [reply docs/release-notes/s3gw-v0.14.0.md]: good catch! fixed

**Review by m-ildefons — APPROVED** [review]
LGTM

**Pull request merged**